core/ArchiveProcessor/Loader.php

```diff
@@ -113,12 +115,19 @@ private function prepareArchiveImpl($pluginName)
             }
         }
 
+        // invalidate existing archives before we start archiving in case data was tracked in the past. if the archive is
+        // made invalid, we will correctly re-archive below.
+        if ($this->invalidateBeforeArchiving) {
+            $this->invalidatedReportsIfNeeded();
+        }
+
         // NOTE: $idArchives will contain the latest DONE_OK/DONE_INVALIDATED archive as well as any partial archives
         // with a ts_archived >= the DONE_OK/DONE_INVALIDATED date.
-        list($idArchives, $visits, $visitsConverted, $isAnyArchiveExists) = $this->loadExistingArchiveIdFromDb();
+        list($idArchives, $visits, $visitsConverted, $isAnyArchiveExists, $tsArchived, $value) = $this->loadExistingArchiveIdFromDb(); // TODO: $value
         if (!empty($idArchives)
             && !$this->params->getArchiveOnlyReport()
             && !Rules::isForceArchivingSinglePlugin()
+            && !$this->shouldForceInvalidatedArchive($value, $tsArchived)
```

Comment:
Note: if an archive is found to be invalidated, we check if we should force archiving. This check was originally in ArchiveSelector, but I moved it since ArchiveSelector is used in multiple places now.